test/index.js

```diff
@@ -4105,6 +4105,81 @@ describe('list', function() {
 
 describe('object', function() {
 
+  describe('prop', function() {
+
+    it('is a binary function', function() {
+      eq(typeof S.prop, 'function');
+      eq(S.prop.length, 2);
+    });
+
+    it('type checks its arguments', function() {
+      throws(function() { S.prop(1); },
+             errorEq(TypeError,
+                     'Invalid value\n' +
+                     '\n' +
+                     'prop :: Accessible a => String -> a -> b\n' +
+                     '                        ^^^^^^\n' +
+                     '                          1\n' +
+                     '\n' +
+                     '1)  1 :: Number, FiniteNumber, NonZeroFiniteNumber, Integer, ValidNumber\n' +
+                     '\n' +
+                     'The value at position 1 is not a member of ‘String’.\n'));
+
+      throws(function() { S.prop('a', null); },
+             errorEq(TypeError,
+                    'Type-class constraint violation\n' +
+                    '\n' +
+                    'prop :: Accessible a => String -> a -> b\n' +
+                    '        ^^^^^^^^^^^^              ^\n' +
+                    '                                  1\n' +
+                    '\n' +
+                    '1)  null :: Null\n' +
+                    '\n' +
+                    '‘prop’ requires ‘a’ to satisfy the Accessible type-class constraint; the value at position 1 does not.\n'));
+
+      assert.throws(function() { S.prop('a', true); },
+                    errorEq(TypeError,
+                            '‘prop’ expected object to have a property ' +
+                            'named ‘a’; true does not'));
+
+      assert.throws(function() { S.prop('a', 1); },
+                    errorEq(TypeError,
+                            '‘prop’ expected object to have a property ' +
+                            'named ‘a’; 1 does not'));
+    });
+
+    it('throws when the property is not present', function() {
+      assert.throws(function() { S.prop('map', 'abcd'); },
+                    errorEq(TypeError,
+                            '‘prop’ expected object to have a property ' +
+                            'named ‘map’; "abcd" does not'));
+
+      assert.throws(function() { S.prop('c', {a: 0, b: 1}); },
+                    errorEq(TypeError,
+                            '‘prop’ expected object to have a property ' +
+                            'named ‘c’; {"a": 0, "b": 1} does not'));
+
+      assert.throws(function() { S.prop('xxx', [1, 2, 3]); },
+                    errorEq(TypeError,
+                            '‘prop’ expected object to have a property ' +
+                            'named ‘xxx’; [1, 2, 3] does not'));
```

Comment:
Let's use `throws` rather than `assert.throws` for these assertions as well.

Also, now that we're not enforcing a maximum line length let's not wrap the messages. :)